### ISSUE TITLE
react/prop-types false-positive when using with flow & and object deconstruct. 

### ISSUE THREAD

**Issue by Yangobears** [yannickcr/eslint-plugin-react#1185]
Using ```&``` to construct Props 
``` 
type PropsOne = {
  a: string, 
}

type PropsTwo = {
  b: string,
}
type Props = PropsOne & PropsTwo;
```
and deconstruct the object
```
renderSth = ({c,d,e}: {c?: string, d?: string, e?: boolean}) => {}
``` 
would give 
```
error  'a' is missing in props validation        react/prop-types
error  'b' is missing in props validation        react/prop-types
```

The issue would not happen if 
explicitly declare the props
```
type Props = {
  a: string, 
  b: string,
}
```
or choosing not to deconstruct the object
```
renderSth = (args: {c?: string, d?: string, e?: boolean}) => {}
```

**Comment by ljharb** [comment]
`prop-types` should no longer trigger on render functions like this.

If you're still seeing issues on components with the latest version of the plugin, please file a new issue.
